packages/sdk/src/cli/checkUnsupportedDataProxy.ts

```diff
@@ -0,0 +1,76 @@
+import chalk from 'chalk'
+import { O } from 'ts-toolbelt'
+
+import { getConfig, getSchemaPath, link } from '..'
+import { loadEnvFile } from '../utils/loadEnvFile'
+
+/**
+ * These are the cli args that we check the data proxy for. If in use
+ */
+const checkedArgs = {
+  '--url': true,
+  '--to-url': true,
+  '--from-url': true,
+  '--shadow-database-url': true,
+  '--schema': true,
+  '--from-schema-datamodel': true,
+  '--to-schema-datamodel': true,
+}
+
+type Args = O.Optional<O.Update<typeof checkedArgs, any, string>>
+
+/**
+ * Get the message to display when a command is forbidden with a data proxy flag
+ * @param command the cli command (eg. db push)
+ * @returns
+ */
+export const forbiddenCmdWithDataProxyFlagMessage = (command: string) => `
+${chalk.green(`prisma ${command}`)} command is not yet available with ${chalk.green('prisma://')}.
+
+More information in our documentation:
+${link('https://pris.ly/d/data-proxy')}
+`
+
+/**
+ * Check that the data proxy cannot be used through the given args
+ * @param command the cli command (eg. db push)
+ * @param args the cli command arguments
+ * @param implicitSchema if this command implicitly loads a schema
+ */
+async function _checkUnsupportedDataProxy(command: string, args: Args, implicitSchema: boolean) {
+  const argList = Object.entries(args)
+
+  // when the schema can be implicit, we use its default location
+  if (implicitSchema === true) {
+    args['--schema'] = (await getSchemaPath(args['--schema'])) ?? undefined
```

Comment:
Similar logic to what we do in the cli when we need to get the location.

Reply:
Does that mean `getSchemaPath` is some logic like 

1. default `prisma/schema.prisma`, 
2. otherwise what is in `--schema`, 
3. otherwise what is defined in `package.json` as `prisma.schema`"`

or something like that, making sure that the same schema file is used that the CLI command would use?

Reply:
Yes, definitely. CLI commands use both `getSchemaPathAndPrint` and `getSchemaPath`.
1. If --schema had a value, check if it exists and get its absolute path
2. If empty, then look in these locations 
	  ```ts
	  (await getSchemaPathFromPackageJson(opts.cwd)) ??
	  (await getRelativeSchemaPath(opts.cwd)) ??
	  (await resolveYarnSchema(opts.cwd))
	  ```